src/entrypoint.sh

```diff
@@ -46,6 +41,10 @@ function ssh_setup() {
 # Forward to /dev/null to swallow the output of the private key
 if [[ -n "${SSH_PRIVATE_KEY_SRC}" ]] &>/dev/null; then
   ssh_setup
+else
+  # gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}"
```

Comment:
not working, if `GITHUB_TOKEN` env variable is set, the command throws.
@msiebeneicher is it possible that you use another `gh` version? I do not understand why this is happening

Reply:
The order here looks mandatory. Also it is not possible to use `gh auth status` here because gh wants to check a github.com auth by default which is not possible in a github enterprise setup. A working solution for my setup is this:

```sh
else
  git config --global "credential.https://${SOURCE_REPO_HOSTNAME}.helper" "!gh auth git-credential"
  gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${GITHUB_TOKEN}"
fi
```

Reply:
related to [docs](https://cli.github.com/manual/gh_auth_status) `gh auth status` should also work on other hosts (GitHub enterprise), but seems not?

Related to your working solution
![Screenshot 2023-01-02 at 17 45 54](https://user-images.githubusercontent.com/8027933/210259291-676d02e3-42f0-44e6-8ba3-3bae34d5b654.png)
n. Sad but it does not work on github.com  https://github.com/AndreasAugustin/actions-template-sync/actions/runs/3823961230/jobs/6505672841

hmmm  if it is not a `gh --version` issue, I guess it is a runner issue

Reply:
weird - for me it's running fine with gh 2.15.0:

![image](https://user-images.githubusercontent.com/5821183/210315720-f135b7ec-5138-4a22-972a-bacc42fe7877.png)

So back to the idea of unsetting the default github_token and using an own env var for it?

Reply:
Yes weird, just tested on my machine within a docker container also with version `2.15.0` (actually within the sync action `2.21.1` is used).

![image](https://user-images.githubusercontent.com/8027933/210317839-2395a36b-c464-4bf5-a1b1-61acb392bab9.png)

It is an alpine image, but also on my arch based system I receive the same error with that small test.

Reply:
aha! maybe the hostname is the issue. Related to https://cli.github.com/manual/gh_help_environment
`GH_TOKEN, GITHUB_TOKEN (in order of precedence): an authentication token for github.com API requests. Setting this avoids being prompted to authenticate and takes precedence over previously stored credentials.`

It states there explicitly `github.com`.

Reply:
**UPDATE**
confirmed! It is the `hostname` variable

Reply:
@msiebeneicher with the changes I just made, the tests are working within github.com. Can you please test them in your env?
Hopefully finally it is a working solution!

Reply:
The last solution works also in my environment 🥳
good catch! Do you want to clean up the code before merge?

Reply:
> The last solution works also in my environment partying_face good catch! Do you want to clean up the code before merge?

think the code is cleaned up, but I will check.
That was a funny and hard issue...